packages/@aws-cdk/aws-s3-deployment/test/test.bucket-deployment.ts

```diff
@@ -239,7 +239,8 @@ export = {
       contentLanguage: "en",
       storageClass: s3deploy.StorageClass.INTELLIGENT_TIERING,
       contentDisposition: "inline",
-      serverSideEncryption: s3deploy.ServerSideEncryption.AES_256,
+      serverSideEncryption: s3deploy.ServerSideEncryption.AWS_KMS,
```

Comment:
This kind of removes our test for the `AES_256` use-case, which we don't want to give up (correct me if i'm missing something here).
Lets just add another test for the `AWS_KMS` case. Which I see is completely missing from this file.

In general, we always prefer adding tests, rather than extending or changing existing ones. Even if it means duplicating most of the test code.

Reply:
Your philosophy might vary, but this is just checking that the code correctly translates the property keys. The actual values of these properties doesn't have any factor on the test. I just changed this to AWS_KMS so I could add an additional property to the assertion. I could have left it as AES_256 and although it doesn't really make sense to specify a KMS key ID along with AES_256, the test would still pass.

Reply:
Another thing to consider is what use case this test is testing: "object metadata can be given" - I haven't eroded this.

Reply:
Your right, I didn't notice which exact test this was in. This test simply validates the given metadata: `{ "A": "1", "b": "2" }` is translated into `{ 'x-amzn-meta-a': '1', 'x-amzn-meta-b': '2' }`.

This means your added properties don't belong there (in fact none of the additional properties belong there). 

What I would expect to see is an another test, perhaps call it 'props translate to system metadata', that validates the properties are translated to the expected keys. This test can than porbably be modified every time we add a property. Its important also not to mix test-cases.

Reply:
I'm sorry, I don't understand this piece of feedback. It seems like you're asserting that this case is testing the metadata -> UserMetadata mapping. However, there are far more fields in this test for everything else -> SystemMetadata in this test. 

Anyway, I'll copy this test case to another test case and remove most of the fields from the original test.

Reply:
Sorry, I just realized the `props translate to system metadata` test im suggesting is in fact just a subset of the `object metadata can be given` test. Its just confusing because `metadata` is the name of property as well. 

I'm still not liking the removal of that property value because in affect it does erode a test. Imagine for some reason we change this [enum](https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts#L298) value, some test should fail. Currently, it looks like its this one.

Reply:
Ok, now the confusion if out of the way, hopefully. 

What i would do is:

- Leave the current test as is.
- Add another test that simply validates translation of the missing keys. Why another test, because I do want to use the AWS_KMS value in this one, but also leave the test for AES256.

Does that make sense?

Reply:
This is a systemic problem with these tests. The following classes have values which are not tested:

* CacheControl
* StorageClass

For the purposes of getting this PR approved, I will add tests for the ServerSideEncryption enum

Reply:
Sounds good

Reply:
I updated this PR with that test.